test/checkImageSize.js

```diff
@@ -0,0 +1,87 @@
+var fixImage = require("./fixImageSize.js");
+const sizeOf = require('image-size');
+const fs = require('fs');
+const { exit } = require("process");
+
+function getFileSizeMB(fileStats){
+    return (fileStats.size) / (1024*1024);
+}
+
+//Check if image is larger than 1MB
+function isFileSizeBig(fileStats){ 
+    return getFileSizeMB(fileStats) > 1;
+}
+
+function getImageWidth(imageSize){
+    return imageSize.width;
+}
+
+function getImageHeight(imageSize){
+    return imageSize.height;
+}
+
+//Check if the width of image is larger than 1024px or lower than 512px
+function isImageWidthUnreasonable(imageSize){
+    let width = imageSize.width;
+    return (width <512 || width > 1024)
+}
+
+function imageSizeChecker(input){  //input here is the image file
+    //let inputName = "./img/" + input;
+    let imageSize = sizeOf(input);
+    let width = imageSize.width;
+    let height = imageSize.height;
+    let imageStats = fs.statSync(input);
+    if (width != height){
+        throw Error("Not a pixel-perfect square aspect ratio")
+    }
+    else{
+        if(isFileSizeBig(imageStats)){
+            throw Error("Your file is " + getFileSizeMB(imageStats).toFixed(2) + "MB")
+        } 
+        else if(isImageWidthUnreasonable(imageSize)){
+            throw Error("Your image size is " + getImageWidth(imageSize) +"px by " +getImageHeight(imageSize)+"px");
```

Comment:
You may want to explain what the error is, ex

* your file should be under 1mb
* your image should be within ... dimensions